hetmech/test_matrix.py

```diff
@@ -66,9 +62,13 @@ def test_metaedge_to_adjacency_matrix(test_edge, exp_row, exp_col, exp_adj,
     row_names, col_names, adj_mat = \
         metaedge_to_adjacency_matrix(graph, test_edge, matrix_type=mat_type,
                                      dtype=dtype)
+    exp_row, exp_col, exp_adj = get_arrays(test_edge, mat_type, dtype)
 
-    assert numpy.array_equal(row_names, exp_row)
-    assert numpy.array_equal(col_names, exp_col)
-    assert not type((exp_adj != adj_mat)).max(exp_adj != adj_mat)
+    assert row_names == exp_row
+    assert col_names == exp_col
+    assert type(adj_mat) == type(exp_adj)  # Arrays same type
+    assert not sparse.csc_matrix(adj_mat != exp_adj).nnz  # Arrays equal
```

Comment:
`adj_mat != exp_adj` gives an array with the element-wise comparison of `adj_mat` and `exp_adj`. If all values are `False`, then the arrays are equal. The problem with using `count_nonzero` is that the matrix type of the comparison array changes depending on the types of matrices under comparison. `sparse.csc_matrix.nnz` is a simply way to check equality, because it gives the number not equal regardless of the types of arrays under review.